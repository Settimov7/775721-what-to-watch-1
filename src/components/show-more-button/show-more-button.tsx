import * as React from 'react';
import {connect} from 'react-redux';

import {ActionCreator} from '../../reducer/films/films';
import {getIsVisibleShowMoreButton} from '../../reducer/films/selectors';

interface Props {
  onClick: () => void;
  isVisible: boolean;
}

export const ShowMoreButton = (props: Props) => {
  const {onClick, isVisible} = props;

  if(!isVisible) {
    return null;
  }

  return (
    <div className="catalog__more">
      <button
        className="catalog__button"
        type="button"
        onClick={(evt) => {
          evt.preventDefault();

          onClick();
        }}
      >
        Show more
      </button>
    </div>
  );
};

const mapStateToProps = (state, ownProps) => Object.assign({}, ownProps, {
  isVisible: getIsVisibleShowMoreButton(state),
});

const mapDispatchToProps = (dispatch) => ({
  onClick: (): void => {
    dispatch(ActionCreator.increaseDisplayedFilmsNumber());
  },
});

export default connect(mapStateToProps, mapDispatchToProps)(ShowMoreButton);
