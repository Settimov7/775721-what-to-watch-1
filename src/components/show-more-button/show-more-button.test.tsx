import * as React from 'react';
import * as renderer from 'react-test-renderer';

import {ShowMoreButton} from './show-more-button';

it(`Show more button correctly renders when is visible`, () => {
  const props = {
    onClick: jest.fn(),
    isVisible: true,
  };

  const showMoreButton = renderer.create(<ShowMoreButton {...props}/>).toJSON();

  expect(showMoreButton).toMatchSnapshot();
});

it(`Show more button correctly renders when is not visible`, () => {
  const props = {
    onClick: jest.fn(),
    isVisible: false,
  };

  const showMoreButton = renderer.create(<ShowMoreButton {...props}/>).toJSON();

  expect(showMoreButton).toMatchSnapshot();
});
