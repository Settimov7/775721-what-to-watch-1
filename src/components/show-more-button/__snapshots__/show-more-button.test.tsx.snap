// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Show more button correctly renders when is visible 1`] = `
<div
  className="catalog__more"
>
  <button
    className="catalog__button"
    onClick={[Function]}
    type="button"
  >
    Show more
  </button>
</div>
`;

exports[`Show more button correctly renders when is not visible 1`] = `null`;
